README.md

```diff
@@ -50,10 +51,60 @@ async function helloWorld (event: WithBody<APIGatewayProxyEvent, NameBody>) {
 
 // Let's "middyfy" our handler, then we will be able to attach middlewares to it
 export const handler = middy(helloWorld)
-  .use(ClassValidatorMiddleware({
-    // Add the validation class here
-    classType: NameBody
-  }))
+  .use(
+    ClassValidatorMiddleware({
+      // Add the validation class here
+      classType: NameBody
+    })
+  )
+  // The class validator throws validation errors from http-errors which are compatible with
+  // the error handler middlewares for middy
+  .use(JSONErrorHandlerMiddleware())
+```
+
+## Usage for query params
+
+```typescript
+// When using decorators, don't forget to import this in the very first line of code
+import 'reflect-metadata'
+
+import { APIGatewayProxyEvent } from 'aws-lambda'
+import { IsString } from 'class-validator'
+import middy from '@middy/core'
+import JSONErrorHandlerMiddleware from 'middy-middleware-json-error-handler'
+
+import ClassValidatorMiddleware, { WithQuery } from 'middy-middleware-class-validator'
+
+// Define a validator for the body via class-validator
+class NameBody {
+  @IsString()
+  public firstName: string
+
+  @IsString()
+  public lastName: string
+}
+
+// This is your AWS handler
+async function helloWorld (event: WithQuery<APIGatewayProxyEvent, NameBody>) {
+  // Thanks to the validation middleware you can be sure queryStringParameters is typed correctly
+  return {
+    queryStringParameters: `Hello ${event.queryStringParameters.firstName} ${event.queryStringParameters.lastName}`,
+    headers: {
+      'content-type': 'text'
+    },
+    statusCode: 200
+  }
+}
+
+// Let's "middyfy" our handler, then we will be able to attach middlewares to it
+export const handler = middy(helloWorld)
+  .use(
+    ClassValidatorMiddleware({
+      // Add the validation class here
+      classType: NameBody,
+      validateQueryParamas: true
```

Comment:
setting a flag is a bit weird. Why do I need to set a flag for validating queryParams, but not for validating the body? How do I set this up if I want to validate both, but differently? 

Two ideas, either of which could work:
1. We export three middlewares: `BodyClassValidatorMiddleware`, `QueryClassValidatorMiddleware` and `ClassValidatorMiddleware` with the last one being a reexport of `BodyClassValidatorMiddleware` that is deprectad with a note to use `BodyClassValidatorMiddleware` instead
2. We offer `bodyClassType`, `queryClassType` and `classType` with the last being a deprecated fallback that ensures none of the other two are set.

What do you think?

Reply:
I opt the second one. I will do the necessary modification.